build.cake

```diff
@@ -5,7 +5,11 @@ using System.Threading.Tasks;
 var target = Argument("target", "Default");
 var exercise = Argument<string>("exercise", null);
 
-var sourceDir = "./exercises";
+var exercisesSourceDir = "./exercises";
```

Comment:
I don't think we need both an `exercisesSourceDir` and `buildDir` variable, as the only thing that is built there are the exercises. I think we could just revert to the old situation here, where the exercise directory are directly placed and built in the `./build` directory. I do think I like your naming better, so I would like to see something like this:

```csharp
var exercisesSourceDir = "./exercises";
var exercisesBuildDir = "./build";
```

And then you could drop the `buildDir` parameter and replace it with `exercisesBuildDir` wherever it is used.